openedx_tagging/core/tagging/models.py

```diff
@@ -1,39 +1,426 @@
 """ Tagging app data models """
+from typing import List, Type
+
 from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+# Maximum depth allowed for a hierarchical taxonomy's tree of tags.
+TAXONOMY_MAX_DEPTH = 3
+
+# Ancestry of a given tag; the Tag.value fields of a given tag and its parents, starting from the root.
+# Will contain TAXONOMY_MAX_DEPTH elements.
+Lineage = List[str]
+
 
+class Tag(models.Model):
+    """
+    Represents a single value in a list or tree of values which can be applied to a particular Open edX object.
 
-class TagContent(models.Model):
+    Open edX tags are "name:value" pairs which can be applied to objects like content libraries, units, or people.
+    Tag.taxonomy.name provides the "name" and the Tag.value provides the "value".
+    (And an ObjectTag links a Tag with an object.)
     """
-    TODO: This is a basic representation of TagContent, it may change
-    depending on the discovery in:
-    https://docs.google.com/document/d/13zfsGDfomSTCp_G-_ncevQHAb4Y4UW0d_6N8R2PdHlA/edit#heading=h.o6fm1hktwp7b
 
-    This is the representation of a tag that is linked to a content
+    id = models.BigAutoField(primary_key=True)
```

Comment:
@ormsbee I'm not clear on when it's better to use an `id` vs a `uuid`, and so I'm not sure which is best as the primary key for these models: Tag, Taxonomy, ObjectTag.

Since we're recommending multi-table inheritance to extend these models, they will share a database. However if we ever move this app to a separate service, I can see how a separate database would prefer to store UUIDs.

What do you think?

Reply:
I think it's okay to have just an `id` here, since the combination of Taxonomy and `external_id` should be enough to uniquely identify a tag. If we find a need for UUIDs, we can add them later.